openshift/console
# Update Devfile Import to use Kubernetes YAML definitions

### Description
This PR updates the Devfile handler backend code to consume the Kubernetes YAML file definitions. For now, the devfile samples only have the `Deployment` definition in the YAML file but the next course of action would be to update the `Service`, `Route` definitions in the devfile sample

Fixes https://github.com/devfile/api/issues/924

This PR just added the ability to read from Kubernetes YAML if the Devfile Kubernetes Component has `inlined` definition. If inlined is not present, it resorts to the previous way of generating Kubernetes resources. This should be compatible with previous OpenShift versions and the master branch

This PR also refactored some of the devfile backend code to keep it more clean and organized. Added tests to the devfile pkg since ODC didnt have go tests. Coverage %:

<img width="1331" alt="Screen Shot 2022-09-22 at 1 33 59 PM" src="https://user-images.githubusercontent.com/31771087/191813939-002f743c-ac5b-4e89-9837-21c4e712d27d.png">


### How to test
(Check history for prev edit)
- Devfile Import should work fine from the UI
- If you are interested in testing the new approach, go to devfile-handler.go and update the following lines. Build and test it out by only importing a _python devfile sample_
```
         // Get devfile content and parse it using a library call in the future
	// devfileContentBytes := []byte(data.Devfile.DevfileContent)
	convert := true
	devfileObj, _, err = devfile.ParseDevfileAndValidate(parser.ParserArgs{
		// Data: devfileContentBytes,
		URL: "https://raw.githubusercontent.com/devfile-samples/devfile-sample-python-basic/87e269d5e018c904ba0a72b53292b3a9ed0b8b37/devfile.yaml",
		ConvertKubernetesContentInUri: &convert,
	})
```
